specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-10-01-preview/SourceControls.json

```diff
@@ -0,0 +1,1007 @@
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "Version",
+        "values": [
+          {
+            "value": "V1"
+          },
+          {
+            "value": "V2"
+          }
+        ]
+      }
+    },
+    "DeploymentFetchStatus": {
+      "description": "Status while trying to fetch the deployment information.",
+      "enum": [
+        "Success",
+        "Unauthorized",
+        "NotFound"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "DeploymentFetchStatus",
+        "values": [
+          {
+            "value": "Success"
+          },
+          {
+            "value": "Unauthorized"
+          },
+          {
+            "value": "NotFound"
+          }
+        ]
+      }
+    },
+    "DeploymentState": {
+      "description": "The current state of the deployment.",
+      "enum": [
+        "In_Progress",
+        "Completed",
+        "Queued",
+        "Canceling"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "DeploymentState",
+        "values": [
+          {
+            "value": "In_Progress"
+          },
+          {
+            "value": "Completed"
+          },
+          {
+            "value": "Queued"
+          },
+          {
+            "value": "Canceling"
+          }
+        ]
+      }
+    },
+    "DeploymentResult": {
+      "description": "Status while trying to fetch the deployment information.",
+      "enum": [
+        "Success",
+        "Canceled",
+        "Failed"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "DeploymentResult",
+        "values": [
+          {
+            "value": "Success"
+          },
+          {
+            "value": "Canceled"
+          },
+          {
+            "value": "Failed"
+          }
+        ]
+      }
+    },
+    "PullRequest": {
+      "readOnly": true,
+      "description": "Information regarding pull request for protected branches.",
+      "type": "object",
+      "properties": {
+        "url": {
+          "description": "URL of pull request",
+          "type": "string",
+          "readOnly": true
+        },
+        "state": {
+          "$ref": "#/definitions/PullRequestState",
+          "description": "State of the pull request",
+          "type": "object",
+          "readOnly": true
+        }
+      }
+    },
+    "PullRequestState": {
+      "description": "Status of the pull request.",
+      "enum": [
+        "Open",
+        "Closed"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "state",
+        "values": [
+          {
+            "value": "Open"
+          },
+          {
+            "value": "Closed"
+          }
+        ]
+      }
+    },
+    "Warning": {
+      "description": "Warning response structure.",
+      "x-ms-external": true,
```

Comment:
Why is this defined as external?

Reply:
It should not be, it is an error done in previous version and was not fixed here. Great catch. @xuhumsft will you be able to change it to `false`?

Reply:
fix in https://github.com/Azure/azure-rest-api-specs/pull/26647. Waiting for merging to the protected branch